devito/finite_differences/differentiable.py

```diff
@@ -47,11 +48,47 @@ def time_order(self):
         return min([getattr(i, 'time_order', 100) or 100 for i in self._args_diff],
                    default=100)
 
+    @cached_property
+    def is_TimeDependent(self):
+        # Default False, True if anything is time dependant in the expression
+        return any(getattr(i, 'is_TimeDependent', False) for i in self._args_diff)
+
+    @cached_property
+    def is_VectorValued(self):
+        # Default False, True if is a vector valued expression
+        return any(getattr(i, 'is_VectorValued', False) for i in self._args_diff)
+
+    @cached_property
+    def is_TensorValued(self):
+        # Default False, True if is a tensor valued expression
+        return any(getattr(i, 'is_TensorValued', False) for i in self._args_diff)
+
+    @cached_property
+    def grid(self):
+        grids = {getattr(i, 'grid', None) for i in self._args_diff} - {None}
+        if len(grids) > 1:
+            warning("Expression contains multiple grids, returning first found")
+        try:
+            return grids.pop()
+        except KeyError:
+            raise ValueError("No grid found")
+            return None
```

Comment:
drop

Reply:
why?

Reply:
well you're raising an exception just before it so it will never be reached !